packages/shared/src/hooks/useOnPostClick.ts

```diff
@@ -0,0 +1,65 @@
+import { useContext } from 'react';
+import useIncrementReadingRank from './useIncrementReadingRank';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import { feedAnalyticsExtra, postAnalyticsEvent } from '../lib/feed';
+import { Post } from '../graphql/posts';
+import { Origin } from '../lib/analytics';
+
+interface PostClickOptionalProps {
+  skipPostUpdate?: boolean;
+}
+
+export type FeedPostClick = ({
+  event,
+  post,
+  row,
+  column,
+  optional,
+}: {
+  event?: string;
+  post: Post;
+  row?: number;
+  column?: number;
+  optional?: PostClickOptionalProps;
+}) => Promise<void>;
+
+interface UseOnPostClickProps {
+  columns?: number;
+  feedName?: string;
+  ranking?: string;
+  origin?: Origin;
+}
+export default function useOnPostClick({
+  columns,
+  feedName,
+  ranking,
+  origin,
+}: UseOnPostClickProps): FeedPostClick {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { incrementReadingRank } = useIncrementReadingRank();
+
+  return async ({
```

Comment:
`useMemo` might be helpful here I think

Reply:
@idoshamun Added useMemo to this one and `useFeedOnPostClick` hooks.
Also uniformed the event name :)